src/events/http2.js

```diff
@@ -0,0 +1,171 @@
+/**
+ * @fileoverview Handlers for http & https libraries instrumentation
+ */
+
+const uuid4 = require('uuid4');
+const shimmer = require('shimmer');
+const urlLib = require('url');
+const utils = require('../utils.js');
+const tracer = require('../tracer.js');
+const serverlessEvent = require('../proto/event_pb.js');
+const eventInterface = require('../event.js');
+const errorCode = require('../proto/error_code_pb.js');
+const { isBlacklistURL, isBlacklistHeader } = require('.././helpers/events');
+const {
+    resolveHttpPromise,
+    isURLIgnoredByUser,
+    USER_AGENTS_BLACKLIST,
+    URL_BLACKLIST,
+    generateEpsagonTraceId,
+    updateAPIGateway,
+} = require('./http');
+const tryRequire = require('../try_require');
+
+const http2 = tryRequire('http2');
+
+
+/**
+ * Removing ':' from http2 headers.
+ * @param {object} headers data
+ * @returns {object} only real headers without ':'
+ */
+function extractHeaders(headers) {
+    return Object.entries(headers)
+        .filter(header => !header[0].startsWith(':'))
+        .reduce((obj, header) => {
+            const [key, value] = header;
+            obj[key] = value; // eslint-disable-line no-param-reassign
+            return obj;
+        }, {});
+}
+
+/**
+ * Wraps the http2's module request function with tracing
+ * @param {Function} wrappedFunction The http2's request module
+ * @param {string} authority hostname
+ * @returns {Function} The wrapped function
+ */
+function httpWrapper(wrappedFunction, authority) {
+    return function internalHttpWrapper(headers, options) {
+        let clientRequest = null;
+        try {
+            const { hostname } = urlLib.parse(authority);
+
+            const reqHeaders = extractHeaders(headers);
+
+            if (isBlacklistURL(hostname, URL_BLACKLIST, headers[':path']) || isURLIgnoredByUser(hostname)) {
```

Comment:
why are you checking `headers` if you just extracted `reqHeaders`? 
anyway these 2 tests should come before parsing the hostname and extracting the headers

Reply:
For these 2 tests we need - path, hostname, and headers ("cleaned" ones). I'm not sure what is the problem here.